src/modules/Session/model.ts

```diff
@@ -0,0 +1,108 @@
+import { inputObjectType, objectType } from 'nexus/dist';
+import { PrismaClient, Prisma } from '@prisma/client';
+import { randomUUID } from 'crypto';
+import { ApolloError } from 'apollo-server-core';
+import { token } from '../../helpers';
+
+export const Session = objectType({
+  name: 'Session',
+  definition(t) {
+    t.nonNull.string('id');
+    t.nonNull.date('createdAt');
+    t.nonNull.string('createdBy');
+    t.date('referenceExpiryDate');
+    t.nonNull.string('referenceTokenId');
+    t.nonNull.boolean('isUserCreated');
+    t.string('name');
+    t.date('revokedAt');
+  },
+});
+
+export const SessionCreate = inputObjectType({
+  name: 'SessionCreate',
+  definition(t) {
+    t.date('referenceExpiryDate');
```

Comment:
The original suggestion was to accept number of seconds, otherwise you need to validate past date, figure out timezone difference between frontend and server, etc

Also, the idea behind `reference*` prefix is to signal that the value stored in the database is not actually used to validate token (so editing it will not make tokens work longer or shorter). Using this prefix for `input` is confusing